### PR TITLE
dedup download_missing() calls

### PR DESCRIPTION
**Pull request by problame**

### CODE REVIEW

File: pageserver/src/tenant/timeline.rs [hunk 1]
```diff
@@ -1155,29 +1155,31 @@ impl Timeline {
...
-                        .download_missing(&index_part, remote_client, local_filenames, first_save)
-                        .await?;
-                    remote_client.init_upload_queue(&index_part)?;
-                    (local_only_filenames, up_to_date_metadata)
-                }
-                Err(DownloadError::NotFound) => {
-                    info!("no index file was found on the remote");
-                    remote_client.init_upload_queue_empty(&local_metadata)?;
-                    (local_filenames, local_metadata)
+                    _downloaded_index_part = Some(index_part);
```
Comment (LizardWizzard) [inline pageserver/src/tenant/timeline.rs:1163]: In one of the intermediate steps I used Cow to combine reference with an owned value without temporary variables. May look a bit prettier, wdyt?

**Comment by LizardWizzard** [comment]
Thanks for cleaning this up!

File: pageserver/src/tenant/timeline.rs (hunk 1 above)
Comment (problame) [reply pageserver/src/tenant/timeline.rs:1163]: Good idea, forgot that it exists. Will update this PR

**Comment by problame** [comment]
Ready for merge?

**Comment by LizardWizzard** [comment]
Merged manually, thanks!

**Pull request merged**